Upgrade Agent to Collector 0.87 (#5529)
grafana/agent: docs/sources/flow/reference/components/otelcol.receiver.kafka.md

When `mechanism` is set to `"AWS_MSK_IAM"`, the [`aws_msk` child block][aws_msk] must also be provided.

> The `version` argument can be set to either `0` or `1`.

Reviewer: ```suggestion
You can set the `version` argument to `0` or `1`.
```

> to `false` can block the entire Kafka partition if message processing returns
> a permanent error, such as failing to decode.

### header_extraction block

The `header_extraction` block configures how to extract headers from Kafka records.

The following arguments are supported:

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`extract_headers` | `bool` | Enables attaching header fields to resource attributes. | `false` | no

> `headers` | `list(string)` | A list of headers to extract from the Kafka record. | `[]]` | no

Reviewer 2: ```suggestion
`headers` | `list(string)` | A list of headers to extract from the Kafka record. | `[]` | no
```
